From 452f1bfe10472dff38434930ed84c27ee8d4b5a3 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:51 -0400
Subject: [ethernet] mlx4: Removed unnecessary bit operation condition

Message-id: <d1b2b66d4a60f534462ddf7cae907831d369b085.1410425016.git.avadai@redhat.com>
Patchwork-id: 93657
O-Subject: [Patch RHEL7.1 V1 115/197] net/mlx4_core: Removed unnecessary bit operation condition
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eyal Perry <eyalpe@mellanox.com>

Fix the "warn: suspicious bitop condition" made by the smatch semantic
checker on:
drivers/net/ethernet/mellanox/mlx4/main.c:509 mlx4_slave_cap()

Signed-off-by: Eyal Perry <eyalpe@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 483e01320eee45a32061c874a09cbac275effe24)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index 0a03150..526cca8 100644
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -104,8 +104,6 @@ module_param(enable_64b_cqe_eqe, bool, 0444);
 MODULE_PARM_DESC(enable_64b_cqe_eqe,
    "Enable 64 byte CQEs/EQEs when the FW supports this (default: True)");
 
-#define HCA_GLOBAL_CAP_MASK            0
-
 #define PF_CONTEXT_BEHAVIOUR_MASK MLX4_FUNC_CAP_64B_EQE_CQE
 
 static char mlx4_version[] =
@@ -582,9 +580,10 @@ static int mlx4_slave_cap(struct mlx4_dev *dev)
   return err;
  }
 
- /*fail if the hca has an unknown capability */
- if ((hca_param.global_caps | HCA_GLOBAL_CAP_MASK) !=
-     HCA_GLOBAL_CAP_MASK) {
+ /* fail if the hca has an unknown global capability
+  * at this time global_caps should be always zeroed
+  */
+ if (hca_param.global_caps) {
   mlx4_err(dev, "Unknown hca global capabilities\n");
   return -ENOSYS;
  }
-- 
1.7.1